sgt-back-end/index.js

```diff
@@ -0,0 +1,169 @@
+var express = require('express');
+var app = express();
+const pg = require('pg');
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+app.use(express.json());
+
+app.get('/api/grades', (req, res, next) => {
+  const sql = `
+    select "gradeId",
+           "name",
+           "course",
+           "score",
+           "createdAt"
+      from "grades"
+      `;
+  db.query(sql)
+    .then(result => {
+      const getRows = result.rows;
+      res.status(200).json(getRows);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({ error: 'An unexpected error occurred.' });
+    });
+});
+
+app.post('/api/grades', (req, res, next) => {
+
+  const newName = req.body.name;
+  const newCourse = req.body.course;
+  const newScore = req.body.score;
+
+  const scoreNum = parseInt(newScore, 10);
+
+  if (newName === undefined && newCourse !== undefined && newScore !== undefined) {
+    return res.status(400).json({ error: '"name" is a required field.' });
+  } else if (newName !== undefined && newCourse === undefined && newScore !== undefined) {
+    return res.status(400).json({ error: '"course" is a required field.' });
+  } else if (newName !== undefined && newCourse !== undefined && newScore === undefined) {
+    return res.status(400).json({ error: '"score" is a required field.' });
+  } else if (newName === undefined && newCourse === undefined && newScore !== undefined) {
+    return res.status(400).json({ error: '"name" and "course" are required.' });
+  } else if (newName !== undefined && newCourse === undefined && newScore === undefined) {
+    return res.status(400).json({ error: '"course" and "score" are required.' });
+  } else if (newName === undefined && newCourse !== undefined && newScore === undefined) {
+    return res.status(400).json({ error: '"name" and "score" are required.' });
+  } else if (newName !== undefined && newCourse !== undefined && newScore !== undefined) {
+    if (!Number.isInteger(scoreNum)) {
+      return res.status(400).json({ error: '"score" must be an integer.' });
+    } else if (Number.isInteger(scoreNum) && (scoreNum < 0 || scoreNum > 100)) {
+      return res.status(400).json({ error: '"score" must be an integer from 0 to 100.' });
+    }
```

Comment:
Any `score` which isn't an integer should result in a `400 Bad Request`, but `parseInt` converts many non-integers to integers, so stuff like this is currently treated as a valid score, even though it isn't:

---

![image](https://user-images.githubusercontent.com/16143771/132389156-e8102244-f8ba-4456-b756-144166389387.png)

Reply:
Updated to `Number()` .